### ISSUE TITLE
ACL error in server logs when making annotation file

### ISSUE THREAD

**Issue by ppavlidis** [PavlidisLab/Gemma#774]
I just noticed this

```
[Gemma - http-nio-8181-exec-651 (2023-07-17 01:51:04,539)] ERROR ubic.gemma.web.controller.expression.arrayDesign.ArrayDesignControllerImpl.downloadAnnotationFile(205) | Failed to create annotation file /space/gemmaData/microAnnots/GPL6947_bioProcess.an.txt.gz for ArrayDesign Id=503 Name=Illumina HumanHT-12 V3.0 expression bead\
chip Short Name=GPL6947.
java.lang.NullPointerException: null
        at gemma.gsec.acl.domain.AclGrantedAuthoritySid.hashCode(AclGrantedAuthoritySid.java:63) ~[gemma-gsec-0.0.11.jar:?]
        at gemma.gsec.acl.domain.AclEntry.hashCode(AclEntry.java:196) ~[gemma-gsec-0.0.11.jar:?]
        at java.util.HashMap.hash(HashMap.java:340) ~[?:1.8.0_352]
        at java.util.HashMap.put(HashMap.java:613) ~[?:1.8.0_352]
        at java.util.HashSet.add(HashSet.java:220) ~[?:1.8.0_352]
        at java.util.AbstractCollection.addAll(AbstractCollection.java:344) ~[?:1.8.0_352]
        at org.hibernate.collection.internal.PersistentSet.endRead(PersistentSet.java:346) ~[hibernate-core-4.2.21.Final.jar:4.2.21.Final]
        at org.hibernate.engine.loading.internal.CollectionLoadContext.endLoadingCollection(CollectionLoadContext.java:245) ~[hibernate-core-4.2.21.Final.jar:4.2.21.Final]
        at org.hibernate.engine.loading.internal.CollectionLoadContext.endLoadingCollections(CollectionLoadContext.java:235) ~[hibernate-core-4.2.21.Final.jar:4.2.21.Final]
        at org.hibernate.engine.loading.internal.CollectionLoadContext.endLoadingCollections(CollectionLoadContext.java:211) ~[hibernate-core-4.2.21.Final.jar:4.2.21.Final]
        at org.hibernate.loader.Loader.endCollectionLoad(Loader.java:1154) ~[hibernate-core-4.2.21.Final.jar:4.2.21.Final]
        at org.hibernate.loader.Loader.initializeEntitiesAndCollections(Loader.java:1123) ~[hibernate-core-4.2.21.Final.jar:4.2.21.Final]
        at org.hibernate.loader.Loader.processResultSet(Loader.java:969) ~[hibernate-core-4.2.21.Final.jar:4.2.21.Final]
        at org.hibernate.loader.Loader.doQuery(Loader.java:917) ~[hibernate-core-4.2.21.Final.jar:4.2.21.Final]
        at org.hibernate.loader.Loader.doQueryAndInitializeNonLazyCollections(Loader.java:348) ~[hibernate-core-4.2.21.Final.jar:4.2.21.Final]
        at org.hibernate.loader.Loader.doQueryAndInitializeNonLazyCollections(Loader.java:318) ~[hibernate-core-4.2.21.Final.jar:4.2.21.Final]
        at org.hibernate.loader.Loader.loadEntity(Loader.java:2145) ~[hibernate-core-4.2.21.Final.jar:4.2.21.Final]
        at org.hibernate.loader.entity.AbstractEntityLoader.load(AbstractEntityLoader.java:82) ~[hibernate-core-4.2.21.Final.jar:4.2.21.Final]
        at org.hibernate.loader.entity.AbstractEntityLoader.load(AbstractEntityLoader.java:72) ~[hibernate-core-4.2.21.Final.jar:4.2.21.Final]
        at org.hibernate.loader.entity.AbstractEntityLoader.load(AbstractEntityLoader.java:65) ~[hibernate-core-4.2.21.Final.jar:4.2.21.Final]
        at org.hibernate.loader.entity.LegacyBatchingEntityLoaderBuilder$LegacyBatchingEntityLoader.load(LegacyBatchingEntityLoaderBuilder.java:121) ~[hibernate-core-4.2.21.Final.jar:4.2.21.Final]
        at org.hibernate.persister.entity.AbstractEntityPersister.load(AbstractEntityPersister.java:3939) ~[hibernate-core-4.2.21.Final.jar:4.2.21.Final]
        at org.hibernate.event.internal.DefaultLoadEventListener.loadFromDatasource(DefaultLoadEventListener.java:462) ~[hibernate-core-4.2.21.Final.jar:4.2.21.Final]
        at org.hibernate.event.internal.DefaultLoadEventListener.doLoad(DefaultLoadEventListener.java:431) ~[hibernate-core-4.2.21.Final.jar:4.2.21.Final]
        at org.hibernate.event.internal.DefaultLoadEventListener.load(DefaultLoadEventListener.java:208) ~[hibernate-core-4.2.21.Final.jar:4.2.21.Final]
        at org.hibernate.event.internal.DefaultLoadEventListener.proxyOrLoad(DefaultLoadEventListener.java:264) ~[hibernate-core-4.2.21.Final.jar:4.2.21.Final]
        at org.hibernate.event.internal.DefaultLoadEventListener.onLoad(DefaultLoadEventListener.java:152) ~[hibernate-core-4.2.21.Final.jar:4.2.21.Final]
        at org.hibernate.internal.SessionImpl.fireLoad(SessionImpl.java:1053) ~[hibernate-core-4.2.21.Final.jar:4.2.21.Final]
        at org.hibernate.internal.SessionImpl.internalLoad(SessionImpl.java:980) ~[hibernate-core-4.2.21.Final.jar:4.2.21.Final]
        at org.hibernate.type.EntityType.resolveIdentifier(EntityType.java:714) ~[hibernate-core-4.2.21.Final.jar:4.2.21.Final]
        at org.hibernate.type.EntityType.resolve(EntityType.java:502) ~[hibernate-core-4.2.21.Final.jar:4.2.21.Final]
        at org.hibernate.engine.internal.TwoPhaseLoad.doInitializeEntity(TwoPhaseLoad.java:168) ~[hibernate-core-4.2.21.Final.jar:4.2.21.Final]
        at org.hibernate.engine.internal.TwoPhaseLoad.initializeEntity(TwoPhaseLoad.java:137) ~[hibernate-core-4.2.21.Final.jar:4.2.21.Final]
        at org.hibernate.loader.Loader.initializeEntitiesAndCollections(Loader.java:1112) ~[hibernate-core-4.2.21.Final.jar:4.2.21.Final]
        at org.hibernate.loader.Loader.processResultSet(Loader.java:969) ~[hibernate-core-4.2.21.Final.jar:4.2.21.Final]
        at org.hibernate.loader.Loader.doQuery(Loader.java:917) ~[hibernate-core-4.2.21.Final.jar:4.2.21.Final]
        at org.hibernate.loader.Loader.doQueryAndInitializeNonLazyCollections(Loader.java:348) ~[hibernate-core-4.2.21.Final.jar:4.2.21.Final]
        at org.hibernate.loader.Loader.doQueryAndInitializeNonLazyCollections(Loader.java:318) ~[hibernate-core-4.2.21.Final.jar:4.2.21.Final]
        at org.hibernate.loader.Loader.loadEntityBatch(Loader.java:2231) ~[hibernate-core-4.2.21.Final.jar:4.2.21.Final]
        at org.hibernate.loader.entity.LegacyBatchingEntityLoaderBuilder$LegacyBatchingEntityLoader.load(LegacyBatchingEntityLoaderBuilder.java:108) ~[hibernate-core-4.2.21.Final.jar:4.2.21.Final]
        at org.hibernate.persister.entity.AbstractEntityPersister.load(AbstractEntityPersister.java:3939) ~[hibernate-core-4.2.21.Final.jar:4.2.21.Final]
        at org.hibernate.event.internal.DefaultLoadEventListener.loadFromDatasource(DefaultLoadEventListener.java:462) ~[hibernate-core-4.2.21.Final.jar:4.2.21.Final]
        at org.hibernate.event.internal.DefaultLoadEventListener.doLoad(DefaultLoadEventListener.java:431) ~[hibernate-core-4.2.21.Final.jar:4.2.21.Final]
        at org.hibernate.event.internal.DefaultLoadEventListener.load(DefaultLoadEventListener.java:208) ~[hibernate-core-4.2.21.Final.jar:4.2.21.Final]
        at org.hibernate.event.internal.DefaultLoadEventListener.proxyOrLoad(DefaultLoadEventListener.java:264) ~[hibernate-core-4.2.21.Final.jar:4.2.21.Final]
        at org.hibernate.event.internal.DefaultLoadEventListener.onLoad(DefaultLoadEventListener.java:152) ~[hibernate-core-4.2.21.Final.jar:4.2.21.Final]
        at org.hibernate.internal.SessionImpl.fireLoad(SessionImpl.java:1053) ~[hibernate-core-4.2.21.Final.jar:4.2.21.Final]
        at org.hibernate.internal.SessionImpl.internalLoad(SessionImpl.java:980) ~[hibernate-core-4.2.21.Final.jar:4.2.21.Final]
        at org.hibernate.type.EntityType.resolveIdentifier(EntityType.java:714) ~[hibernate-core-4.2.21.Final.jar:4.2.21.Final]
        at org.hibernate.type.EntityType.resolve(EntityType.java:502) ~[hibernate-core-4.2.21.Final.jar:4.2.21.Final]
        at org.hibernate.engine.internal.TwoPhaseLoad.doInitializeEntity(TwoPhaseLoad.java:168) ~[hibernate-core-4.2.21.Final.jar:4.2.21.Final]
        at org.hibernate.engine.internal.TwoPhaseLoad.initializeEntity(TwoPhaseLoad.java:137) ~[hibernate-core-4.2.21.Final.jar:4.2.21.Final]
        at org.hibernate.loader.Loader.initializeEntitiesAndCollections(Loader.java:1112) ~[hibernate-core-4.2.21.Final.jar:4.2.21.Final]
        at org.hibernate.loader.Loader.processResultSet(Loader.java:969) ~[hibernate-core-4.2.21.Final.jar:4.2.21.Final]
        at org.hibernate.loader.Loader.doQuery(Loader.java:917) ~[hibernate-core-4.2.21.Final.jar:4.2.21.Final]
        at org.hibernate.loader.Loader.doQueryAndInitializeNonLazyCollections(Loader.java:348) ~[hibernate-core-4.2.21.Final.jar:4.2.21.Final]
        at org.hibernate.loader.Loader.doList(Loader.java:2550) ~[hibernate-core-4.2.21.Final.jar:4.2.21.Final]
        at org.hibernate.loader.Loader.doList(Loader.java:2536) ~[hibernate-core-4.2.21.Final.jar:4.2.21.Final]
        at org.hibernate.loader.Loader.listIgnoreQueryCache(Loader.java:2366) ~[hibernate-core-4.2.21.Final.jar:4.2.21.Final]
        at org.hibernate.loader.Loader.list(Loader.java:2361) ~[hibernate-core-4.2.21.Final.jar:4.2.21.Final]
        at org.hibernate.loader.hql.QueryLoader.list(QueryLoader.java:495) ~[hibernate-core-4.2.21.Final.jar:4.2.21.Final]
        at org.hibernate.hql.internal.ast.QueryTranslatorImpl.list(QueryTranslatorImpl.java:357) ~[hibernate-core-4.2.21.Final.jar:4.2.21.Final]
        at org.hibernate.engine.query.spi.HQLQueryPlan.performList(HQLQueryPlan.java:198) ~[hibernate-core-4.2.21.Final.jar:4.2.21.Final]
        at org.hibernate.internal.SessionImpl.list(SessionImpl.java:1230) ~[hibernate-core-4.2.21.Final.jar:4.2.21.Final]
        at org.hibernate.internal.QueryImpl.list(QueryImpl.java:101) ~[hibernate-core-4.2.21.Final.jar:4.2.21.Final]
        at gemma.gsec.acl.domain.AclDaoImpl.loadAcls(AclDaoImpl.java:502) ~[gemma-gsec-0.0.11.jar:?]
        at gemma.gsec.acl.domain.AclDaoImpl.readAclsById(AclDaoImpl.java:285) ~[gemma-gsec-0.0.11.jar:?]
        at sun.reflect.GeneratedMethodAccessor12750.invoke(Unknown Source) ~[?:?]
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:1.8.0_352]
        at java.lang.reflect.Method.invoke(Method.java:498) ~[?:1.8.0_352]
        at org.springframework.aop.support.AopUtils.invokeJoinpointUsingReflection(AopUtils.java:317) ~[spring-aop-3.2.18.RELEASE.jar:3.2.18.RELEASE]
        at org.springframework.aop.framework.ReflectiveMethodInvocation.invokeJoinpoint(ReflectiveMethodInvocation.java:183) ~[spring-aop-3.2.18.RELEASE.jar:3.2.18.RELEASE]
        at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:150) ~[spring-aop-3.2.18.RELEASE.jar:3.2.18.RELEASE]
        at org.springframework.dao.support.PersistenceExceptionTranslationInterceptor.invoke(PersistenceExceptionTranslationInterceptor.java:155) ~[spring-tx-3.2.18.RELEASE.jar:3.2.18.RELEASE]
        at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:172) ~[spring-aop-3.2.18.RELEASE.jar:3.2.18.RELEASE]
        at org.springframework.aop.framework.JdkDynamicAopProxy.invoke(JdkDynamicAopProxy.java:204) ~[spring-aop-3.2.18.RELEASE.jar:3.2.18.RELEASE]
        at com.sun.proxy.$Proxy1594.readAclsById(Unknown Source) ~[?:?]
        at gemma.gsec.acl.domain.AclServiceImpl.doReadAcls(AclServiceImpl.java:129) ~[gemma-gsec-0.0.11.jar:?]
        at gemma.gsec.acl.domain.AclServiceImpl.readAclsById(AclServiceImpl.java:112) ~[gemma-gsec-0.0.11.jar:?]
        at sun.reflect.GeneratedMethodAccessor12916.invoke(Unknown Source) ~[?:?]

...

        at org.apache.tomcat.util.net.SocketProcessorBase.run(SocketProcessorBase.java:49) [tomcat-coyote.jar:8.5.71]
        at org.apache.tomcat.util.threads.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1191) [tomcat-util.jar:8.5.71]
        at org.apache.tomcat.util.threads.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:659) [tomcat-util.jar:8.5.71]
        at org.apache.tomcat.util.threads.TaskThread$WrappingRunnable.run(TaskThread.java:61) [tomcat-util.jar:8.5.71]
        at java.lang.Thread.run(Thread.java:750) [?:1.8.0_352]
[Gemma - http-nio-8181-exec-640 (2023-07-17 01:53:11,133)] ERROR ubic.gemma.web.controller.expression.arrayDesign.ArrayDesignControllerImpl.downloadAnnotationFile(205) | Failed to create annotation file /space/gemmaData/microAnnots/GPL6947_bioProcess.an.txt.gz for ArrayDesign Id=503 Name=Illumina HumanHT-12 V3.0 expression bead\
chip Short Name=GPL6947.
java.lang.IllegalStateException: ACLs did not contain key for parent object identity of gemma.gsec.acl.domain.AclObjectIdentity[Type: ubic.gemma.model.expression.experiment.ExpressionExperimentSubSet; Identifier: 6269]( parent = gemma.gsec.acl.domain.AclObjectIdentity[Type: ubic.gemma.model.expression.experiment.ExpressionExper\
iment; Identifier: 6218]); ACLs being inspected: AclImpl[id: 893217; objectIdentity: gemma.gsec.acl.domain.AclObjectIdentity[Type: ubic.gemma.model.expression.experiment.ExpressionExperimentSubSet; Identifier: 6271]; owner: AclPrincipalSid[jhantula]; ^M
AclEntry[id: 3024565; granting: true; sid: AclGrantedAuthoritySid[GROUP_ADMIN]; permission: BasePermission[...........................A....=16]; ]^M
AclEntry[id: 3024566; granting: true; sid: AclGrantedAuthoritySid[GROUP_AGENT]; permission: BasePermission[...............................R=1]; ]^M
AclEntry[id: 3024567; granting: true; sid: AclGrantedAuthoritySid[IS_AUTHENTICATED_ANONYMOUSLY]; permission: BasePermission[...............................R=1]; ]^M
inheriting: true; parent: Null; aclAuthorizationStrategy: gemma.gsec.acl.AclAuthorizationStrategyImpl@4931e5b4; ]
AclImpl[id: 893216; objectIdentity: gemma.gsec.acl.domain.AclObjectIdentity[Type: ubic.gemma.model.analysis.expression.diff.DifferentialExpressionAnalysis; Identifier: 34172]; owner: AclPrincipalSid[jhantula]; no ACEs; inheriting: true; parent: gemma.gsec.acl.domain.AclObjectIdentity[Type: ubic.gemma.model.expression.experiment\
.ExpressionExperimentSubSet; Identifier: 6270]; aclAuthorizationStrategy: gemma.gsec.acl.AclAuthorizationStrategyImpl@4931e5b4; ]
AclImpl[id: 893218; objectIdentity: gemma.gsec.acl.domain.AclObjectIdentity[Type: ubic.gemma.model.analysis.expression.diff.DifferentialExpressionAnalysis; Identifier: 34173]; owner: AclPrincipalSid[jhantula]; no ACEs; inheriting: true; parent: gemma.gsec.acl.domain.AclObjectIdentity[Type: ubic.gemma.model.expression.experiment\
.ExpressionExperimentSubSet; Identifier: 6271]; aclAuthorizationStrategy: gemma.gsec.acl.AclAuthorizationStrategyImpl@4931e5b4; ]
AclImpl[id: 893213; objectIdentity: gemma.gsec.acl.domain.AclObjectIdentity[Type: ubic.gemma.model.expression.experiment.ExpressionExperimentSubSet; Identifier: 6269]; owner: AclPrincipalSid[jhantula]; ^M
AclEntry[id: 3024553; granting: true; sid: AclGrantedAuthoritySid[GROUP_ADMIN]; permission: BasePermission[...........................A....=16]; ]^M
AclEntry[id: 3024554; granting: true; sid: AclGrantedAuthoritySid[GROUP_AGENT]; permission: BasePermission[...............................R=1]; ]^M
AclEntry[id: 3024555; granting: true; sid: AclGrantedAuthoritySid[IS_AUTHENTICATED_ANONYMOUSLY]; permission: BasePermission[...............................R=1]; ]^M
inheriting: true; parent: Null; aclAuthorizationStrategy: gemma.gsec.acl.AclAuthorizationStrategyImpl@4931e5b4; ]
AclImpl[id: 893212; objectIdentity: gemma.gsec.acl.domain.AclObjectIdentity[Type: ubic.gemma.model.analysis.expression.diff.DifferentialExpressionAnalysis; Identifier: 34170]; owner: AclPrincipalSid[jhantula]; no ACEs; inheriting: true; parent: gemma.gsec.acl.domain.AclObjectIdentity[Type: ubic.gemma.model.expression.experiment\
.ExpressionExperimentSubSet; Identifier: 6268]; aclAuthorizationStrategy: gemma.gsec.acl.AclAuthorizationStrategyImpl@4931e5b4; ]
AclImpl[id: 893215; objectIdentity: gemma.gsec.acl.domain.AclObjectIdentity[Type: ubic.gemma.model.expression.experiment.ExpressionExperimentSubSet; Identifier: 6270]; owner: AclPrincipalSid[jhantula]; ^M
AclEntry[id: 3024559; granting: true; sid: AclGrantedAuthoritySid[GROUP_ADMIN]; permission: BasePermission[...........................A....=16]; ]^M
AclEntry[id: 3024560; granting: true; sid: AclGrantedAuthoritySid[GROUP_AGENT]; permission: BasePermission[...............................R=1]; ]^M
```

**Comment by arteymix** [comment]
This is a bug, an AclGrantedAuthoritySid must have a non-null grantedAuthority field. All the fields in the database are non-null.

I'll investigate this tomorrow.

**Comment by arteymix** [comment]
This got fixed in gsec 0.0.16.
